src/main/java/AddCommand.java

```diff
@@ -0,0 +1,70 @@
+import java.io.IOException;
+
+public class AddCommand extends Command {
+
+    public AddCommand(String command) {
+        super(command);
+    }
+
+    boolean isExit() {
+        return false;
+    }
+
+    void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        String arr[] = command.split(" ", 2);
+        String firstWord = arr[0];
+
+        if (firstWord.equals("todo")) {
+            if (arr.length > 1) { //check for errors
```

Comment:
Perhaps you want to abstract out the code chunk inside `if (arr.length > 1) { ... }` for nicer code. You can do the same for other similar code chunk.